compiler/parser/chapel.ypp

```diff
@@ -1084,6 +1089,30 @@ $$ = buildIfStmt(convertAssignmentAndWarn($2,$3,$4), $6, $8); }
 $$ = buildIfStmt(convertAssignmentAndWarn($2,$3,$4), $5, $7); }
 ;
 
+interface_stmt:
+  TINTERFACE ident_def TLP ifc_formal_ls TRP block_stmt
+  { $$ = buildChapelStmt(InterfaceSymbol::buildDef($2, $4, $6)); }
+;
+
+ifc_formal_ls:
+  ifc_formal                      { $$ = new CallExpr(PRIM_ACTUALS_LIST, $1); }
+| ifc_formal_ls TCOMMA ifc_formal { $1->insertAtTail($3); }
+;
+
+ifc_formal:
+  TTYPE ident_def  { $$ = InterfaceSymbol::buildFormal($2, INTENT_TYPE); }
+;
+
+implements_stmt:
+  TIMPLEMENTS ident_def TLP actual_ls TRP TSEMI
+  { $$ = buildChapelStmt(ImplementsStmt::build($2, $4, NULL)); }
+;
+
+implements_expr:
+  TIMPLEMENTS ident_def TLP actual_ls TRP
+  { $$ = ImplementsExpr::build($2, $4); }
```

Comment:
Same as above - I'd like to see `int implements Addable` as an alternative syntax for `implements Addable(int)`.